# Tâche 5
Ce notebook jupyter développe l'effet de 3 éléments de symétrie différents chacun pour un atome différent de la molécule CaN2Cl. Il y sera développé donc l'effet d'un élément de symétrie appliqué à un atome de calcium, d'un élément de symétrie appliqué à un atome d'azote et d'un élément de symétrie appliqué à un atome de chlore.

In [13]:
from mp_api.client import MPRester
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from jupyter_jsmol.pymatgen import quick_view

In [14]:
mp_key = "laRpenx9s6Ae6IiAe8AHZRRyzI5fduqF"
mp_id = "mp-22936"

In [15]:
with MPRester(mp_key) as m:
    prim_struc = m.get_structure_by_material_id(mp_id)

conv_struc = SpacegroupAnalyzer(prim_struc).get_conventional_standard_structure()
symmops = SpacegroupAnalyzer(conv_struc).get_space_group_operations()
sites = conv_struc.sites


Retrieving MaterialsDoc documents:   0%|          | 0/1 [00:00<?, ?it/s]

In [16]:
view = quick_view(conv_struc)
display(view)
view.script('select calcium ; color  gold')
view.script('select nitrogen ; color red')
view.script('select chlorine ; color palegreen')



JsmolView(layout=Layout(align_self='stretch', height='400px'))

Sur cette représentation de la structure cristalline du Ca2NCl, on peut voir les sites où l'on retrouve un atome. En couleur or, les sites où se trouvent les atomes de calcium, en rouge les atomes d'azote et en vert clair, les atomes de chlore

Ci-après est affiché les sites (les positions des atomes dans la structure cristalline) pour chaque site, il est précisé l'atome présent à cette position, les coordonnées cartésiennes et les coordonnées réduites (exprimées par rapport aux vecteurs de base).

In [17]:
print(sites)

[PeriodicSite: Ca (1.829, -1.056, 2.061) [0.6667, 0.3333, 0.1045], PeriodicSite: Ca (0.0, 0.0, 4.515) [0.0, 0.0, 0.2289], PeriodicSite: Ca (1.829, 1.056, 8.636) [0.3333, 0.6667, 0.4378], PeriodicSite: Ca (1.829, -1.056, 11.09) [0.6667, 0.3333, 0.5622], PeriodicSite: Ca (0.0, 0.0, 15.21) [0.0, 0.0, 0.7711], PeriodicSite: Ca (1.829, 1.056, 17.67) [0.3333, 0.6667, 0.8955], PeriodicSite: N (1.829, 1.056, 3.288) [0.3333, 0.6667, 0.1667], PeriodicSite: N (3.658, -4.441e-16, 9.863) [1.0, 1.0, 0.5], PeriodicSite: N (1.829, -1.056, 16.44) [0.6667, 0.3333, 0.8333], PeriodicSite: Cl (0.0, 0.0, 0.0) [0.0, 0.0, 0.0], PeriodicSite: Cl (1.829, -1.056, 6.575) [0.6667, 0.3333, 0.3333], PeriodicSite: Cl (1.829, 1.056, 13.15) [0.3333, 0.6667, 0.6667]]


# Symétrie1
Pour cette première symétrie, prenons un atome de calcium (Ca) par exemple  celui situé en  [0.6667, 0.3333, 0.1045]  exprimé en coordonnées réduites et appliquons une opération de symétrie autre que l'unité :

In [18]:
calc = sites[0]
operation1 = symmops[1]

print("opération de symétrie")
print(symmops[1])

pos_initial_1 = calc.frac_coords
pos_final_1 = operation1.operate(pos_initial_1)

print("Position de l'atome de calcium avant symétrie:",pos_initial_1)
print("Position de l'atome de calcium après symétrie:",pos_final_1)

view1 = quick_view(conv_struc)
display(view1)

view1.script('select calcium ; color  gold')
view1.script('select nitrogen ; color red')
view1.script('select chlorine ; color palegreen')

view1.script('draw SYMOP ' + str(2) + ' {atomno = ' + str(1) + '}')

opération de symétrie
Rot:
[[-1.  0.  0.]
 [ 0. -1.  0.]
 [ 0.  0. -1.]]
tau
[0. 0. 0.]
Position de l'atome de calcium avant symétrie: [0.66666667 0.33333333 0.10445975]
Position de l'atome de calcium après symétrie: [-0.66666667 -0.33333333 -0.10445975]


JsmolView(layout=Layout(align_self='stretch', height='400px'))

### Observation et analyse 
On observe que le premier élément de symétrie appliqué à un atome de calcium transforme les coordonnées (x,y,z) en coordonnées (-x,-y-z). Il n'y a pas de translation. L'opération de symétrie correspond à une inversion. Cette opération de symétrie peut se représenter via le calcul matriciel suivant :

$$\begin{pmatrix} -1 & 0 & 0 \\ 0 & -1 & 0 \\ 0 & 0 & -1 \end{pmatrix}
\begin{pmatrix} 
x \\ 
y \\ 
z
\end{pmatrix} = 
\begin{pmatrix}
- x \\ 
- y \\ 
- z
\end{pmatrix}$$

C'est une opération de deuxième espèce qui inverse donc la chiralité. Cela se caractérise par un déterminant valant -1 ce qui est bien le cas ici.




# Symétrie 2
pour le chlore (Cl), faisons la même chose. On prend par exemple l'atome de chlore situé en [0.33333333 0.66666667 0.66666667] et on lui applique une opération de symétrie. 

In [19]:
cl = sites[11]
operation2 = symmops[7]

print("opération de symétrie")
print(symmops[7])

pos_initial_2 = cl.frac_coords
pos_final_2 = operation2.operate(pos_initial_2)

print("Position de l'atome de chlore avant symétrie:",pos_initial_2)
print("Position de l'atome de chlore après symétrie:",pos_final_2)

view2 = quick_view(conv_struc)
display(view2)
view2.script('select calcium ; color  gold')
view2.script('select nitrogen ; color red')
view2.script('select chlorine ; color palegreen')
view2.script('draw SYMOP ' + str(8) + ' {atomno = ' + str(12) + '}')

opération de symétrie
Rot:
[[ 0. -1.  0.]
 [-1.  0.  0.]
 [ 0.  0.  1.]]
tau
[0. 0. 0.]
Position de l'atome de chlore avant symétrie: [0.33333333 0.66666667 0.66666667]
Position de l'atome de chlore après symétrie: [-0.66666667 -0.33333333  0.66666667]


JsmolView(layout=Layout(align_self='stretch', height='400px'))

### Observation et analyse
On observe que le deuxième élément de symétrie appliqué à un atome de chlore transforme les coordonnées (x,y,z) en coordonnées (-y,-x, z). Cette opération de symétrie correspond à une réflexion. C'est une opération qui inverse la chiralité et donc le déterminant doit valoir -1 ce qui est bien le cas. Cela peut se représenter par le calcul matriciel suivant : 

$$\begin{pmatrix} 0 & -1 & 0 \\ -1 & 0 & 0 \\ 0 & 0 & 1 \end{pmatrix}
\begin{pmatrix} 
x \\ 
y \\ 
z
\end{pmatrix} = 
\begin{pmatrix}
 -y \\ 
 -x \\ 
 z
\end{pmatrix}$$

Il n'y a pas de translation. Cela peut s'observer sur la représentation ci-dessus

# Symétrie 3 
Pour l'azote (N), on réalise la même chose. on prend par exemple les coordonnées réduites suivantes [0.3333, 0.6667, 0.1667] où il y a un atome d'azote.

In [20]:
N = sites[6]
operation3 = symmops[6]

print("opération de symétrie")
print(symmops[6])

pos_initial_3 = N.frac_coords
pos_final_3 = operation3.operate(pos_initial_3)

print("Position de l'atome d'azote avant symétrie:",pos_initial_3)
print("Position de l'atome d'azote après symétrie:",pos_final_3)

view3 = quick_view(conv_struc)
display(view3)
view3.script('select calcium ; color  gold')
view3.script('select nitrogen ; color red')
view3.script('select chlorine ; color palegreen')
view3.script('draw SYMOP ' + str(7) + ' {atomno = ' + str(7) + '}')

opération de symétrie
Rot:
[[ 0.  1.  0.]
 [ 1.  0.  0.]
 [ 0.  0. -1.]]
tau
[0. 0. 0.]
Position de l'atome d'azote avant symétrie: [0.33333333 0.66666667 0.16666667]
Position de l'atome d'azote après symétrie: [ 0.66666667  0.33333333 -0.16666667]


JsmolView(layout=Layout(align_self='stretch', height='400px'))

### Observation et analyse 
On observe que le troisième élément de symétrie appliqué à un atome d'azote transforme les coordonnées (x,y,z) en coordonnées (y,x,-z). Cette troisième symétrie combine plusieurs choses : d'abord on observe une réflexion par rapport au plan x = y caractérisée par l'échange des coordonnées de x et y. Ensuite, on observe une seconde opération de symétrie, une réflexion par rapport au plan xy ce qui inverse z.Ces opérations de symétrie peuvent être représentées par ce calcul matriciel :
$
\begin{pmatrix}
0 & 1 & 0 \\
1 & 0 & 0 \\
0 & 0 & -1
\end{pmatrix}
\begin{pmatrix}
x \\
y \\
z
\end{pmatrix}
=
\begin{pmatrix}
y \\
x \\
- z
\end{pmatrix} 

$$\begin{pmatrix} 0 & 1 & 0 \\ 1 & 0 & 0 \\ 0 & 0 & -1 \end{pmatrix}
\begin{pmatrix} 
x \\ 
y \\ 
z
\end{pmatrix} = 
\begin{pmatrix}
 y \\ 
 x \\ 
 -z
\end{pmatrix}$$

C'est une opération de première espèce qui conserve la chiralité (déterminant = 1). Il n'y a pas de translation. 